apps/emqx_gateway/src/emqx_gateway_http.erl

```diff
@@ -251,10 +251,14 @@ authn(GwName, ListenerId) ->
     {_, Type, Name} = emqx_gateway_utils:parse_listener_id(ListenerId),
     Path = [gateway, GwName, listeners, Type, Name, ?AUTHN],
     ChainName = emqx_gateway_utils:listener_chain(GwName, Type, Name),
-    wrap_chain_name(
-        ChainName,
-        emqx_utils_maps:jsonable_map(emqx:get_raw_config(Path))
-    ).
+    try
+        wrap_chain_name(
+            ChainName,
+            emqx_utils_maps:jsonable_map(emqx:get_raw_config(Path))
+        )
+    catch
+        _:{config_not_found, _} -> authn(GwName)
```

Comment:
We can't just return a Gateway-level authenticator as a Listener-level authenticator. That's not correct.

Reply:
OK